File: php/basics/fibonacci.php
<?php
// Print the first 20 Fibonacci numbers from a generator.

declare(strict_types=1);

function fibonacci(int $count): Generator
{
    $current = 0;
    $next = 1;
    for ($i = 0; $i < $count; $i++) {
        yield $current;
        [$current, $next] = [$next, $current + $next];
    }
}

echo implode(' ', iterator_to_array(fibonacci(20))), PHP_EOL;

File: nim/basics/fibonacci.nim
# Print the first 20 Fibonacci numbers from an inline iterator.
# Run: nim c -r fibonacci.nim

import std/[sequtils, strutils]

iterator fibonacci(count: int): int =
  var current = 0
  var next = 1
  for _ in 1 .. count:
    yield current
    let following = current + next
    current = next
    next = following

echo toSeq(fibonacci(20)).mapIt($it).join(" ")
